### PR TITLE
CNN Conv-layer

### PR DESCRIPTION
**Pull request by goldmermaid**

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-zh/PR-589/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/3 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/6 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/8 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

**Comment by mli** [comment]
Job d2l-zh/PR-589/9 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 1]
```diff
@@ -1,18 +1,22 @@
 # 图像卷积
 :label:`sec_conv_layer`
 
-现在我们了解卷积层在理论上是如何工作的，我们已经准备好了解它们在实践中的工作原理。基于卷积神经网络作为探索图像数据结构的高效架构的动机，我们坚持以图像作为我们的运行例子。
+上节我们解析了卷积层的理论原理，现在我们看看它的实际应用。由于卷积神经网络是人类探索图像数据的结晶，本节我们将以图像为例。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:4]: 理论、原理重复
卷积神经网络的设计是用于探索图像数据。

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 
-## 互相关操作
 
-回想一下，严格来说，卷积层是一个错误的词语，因为它们表达的操作更准确地描述为交叉关联。根据我们在 :numref:`sec_why-conv` 中对卷积层的描述，在这样一个层中，输入张量和核张量结合在一起，通过互相关运算生成输出张量。
+## 互相关运算
 
-让我们现在忽略渠道，看看这是如何处理二维数据和隐藏表示的。在 :numref:`fig_correlation` 中，输入是一个二维张量，高度为 3，宽度为 3。我们将张量的形状标记为 $3 \times 3$ 或者（$3$、$3$）。内核的高度和宽度均为 2。* 内核窗口 *（或 * 卷积窗口 *）的形状由内核的高度和宽度（这里是 $2 \times 2$）给出。
+严格地说，卷积层所表达的运算可以被更准确地描述为 *互相关运算* (cross-correlation)。
+基于 :numref:`sec_why-conv` 中卷积层的描述，在这样一个层中，输入张量和核张量通过互相关运算产生输出张量。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:10]: 这样一个层->卷积层

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 
-![Two-dimensional cross-correlation operation. The shaded portions are the first output element as well as the input and kernel tensor elements used for the output computation: $0\times0+1\times1+3\times2+4\times3=19$.](../img/correlation.svg)
+首先，看看如何处理二维图像数据和隐藏表示。在 :numref:`fig_correlation` 中，输入是高度为 $3$ 、宽度为 $3$ 的二维张量（即 $3 \times 3$ ）。内核的高度和宽度都是 $2$ ，而内核窗口（或卷积窗口）的形状由内核的高度和宽度决定（即 $2 \times 2$ ）。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:12]: +我们暂时忽略通道这一情况。
内核->卷积核

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
+
+![二维互相关运算。阴影部分是第一个输出元素，以及用于计算这个输出的输入和核张量元素: $0\times0+1\times1+3\times2+4\times3=19$.](../img/correlation.svg)
 :label:`fig_correlation`
 
-在二维互相关运算中，我们从位于输入张量左上角的卷积窗口开始，然后从左到右和从上到下滑过输入张量。当卷积窗口滑动到某个位置时，该窗口中包含的输入子张量和核张量将按元素乘以，并将得到的张量总和得出单个标量值。此结果给出相应位置的输出张量值。在这里，输出张量的高度为 2，宽度为 2，四个元素是从二维交叉相关运算得出的：
+在二维互相关运算中，卷积窗口从输入张量的左上角开始，从左到右、从上到下滑动。
+当卷积窗口滑动到新一个位置时，包含在该窗口中的输入子传感器与卷积核张量进行元素相乘，得到的张量相加为一个单一的标量值，由此我们得出了这一位置的输出张量值。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:18]: 输入子传感器->部分张量
进行元素相乘->进行逐元素相乘
得到的张量相加为一个单一的标量值->得到的张量再求和得到一个单一的标量值。

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 5]
```diff
@@ -1,18 +1,22 @@
...
+在如上例子中，输出张量的四个元素由二维互相关运算导出，这个输出高度为 $2$ 、宽度为  $2$ ，如下所示：
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:19]: 导出->得到

**Comment by xiaotinghe** [comment]
还没看完……

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 6]
```diff
@@ -144,22 +155,24 @@ X[:, 2:6].assign(tf.zeros(X[:, 2:6].shape))
 X
 ```
 
-接下来，我们将构建一个高度为 1，宽度为 2 的内核 `K`。当我们对输入执行互相关运算时，如果水平相邻元素相同，则输出为 0。否则，输出为非零。
+接下来，我们构造一个高度为 $1$ 、宽度为 $2$ 的内核 `K` 。当进行互相关运算时，如果水平相邻的两元素相同，则输出为零，否则输出为非零。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:158]: 内核->卷积核。下同

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 7]
```diff
@@ -144,22 +155,24 @@ X[:, 2:6].assign(tf.zeros(X[:, 2:6].shape))
...
 
 ```{.python .input}
 #@tab all
 K = d2l.tensor([[1.0, -1.0]])
 ```
 
-我们已准备好使用参数 `X`（我们的输入）和 `K`（我们的内核）执行交叉相关操作。正如您所看到的，我们检测到 1 表示从白色到黑色的边缘，从黑色到白色的边缘为-1。所有其他输出的值为 0。
+现在，我们对参数 `X` （输入）和 `K` （内核）执行互相关运算。如下所示，输出 `Y` 显示 $1$ 代表从白色到黑色的边缘， $-1$ 代表从黑色到白色的边缘，其他所有输出为 $0$ 。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:165]: 输出 `Y` 中的$1$ 代表从白色到黑色的边缘， $-1$ 代表从黑色到白色的边缘，其他情况的输出为 $0$

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 8]
```diff
@@ -144,22 +155,24 @@ X[:, 2:6].assign(tf.zeros(X[:, 2:6].shape))
...
 
 ```{.python .input}
 #@tab all
 Y = corr2d(X, K)
 Y
 ```
 
-我们现在可以将内核应用于转置的图像。正如预期的那样，它消失了。内核 `K` 只检测垂直边缘。
+现在我们将二维图像输入转置，进行如上的互相关运算。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:173]: 二维图像输入转置->存在二义性。考虑改成 将输入的二维图像转置，再进行如上的互相关运算

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 9]
```diff
@@ -144,22 +155,24 @@ X[:, 2:6].assign(tf.zeros(X[:, 2:6].shape))
...
+其输出如下，之前检测到的垂直边缘消失了。
+不出所料，这个内核 `K` 只可以检测垂直边缘，而非水平边缘。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:175]: 而非->无法检测

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 10]
```diff
@@ -168,19 +181,19 @@ corr2d(d2l.transpose(X), K)
 
 ## 学习内核
 
-如果我们知道这正是我们正在寻找的，通过有限差异 `[1, -1]` 设计边缘检测器是整洁的。但是，当我们查看较大的内核并考虑连续的卷积层时，可能不可能准确地指定每个过滤器应该手动执行的操作。
+如果我们只需寻找黑白边缘，那么以上 `[1, -1]` 的边缘检测器足以。然而，当有了更复杂数值的内核，或者连续的卷积层时，我们不可能手动设计过滤器。那么我们是否可以学习以  `X` 生成 `Y` 的卷积内核呢？
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:184]: 由X生成Y

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 11]
```diff
@@ -168,19 +181,19 @@ corr2d(d2l.transpose(X), K)
...
 
-现在让我们看看是否可以通过仅查看输入-输出对来了解从 `X` 生成 `Y` 的内核。我们首先构建一个卷积层，并将其内核初始化为随机张量。接下来，在每次迭代中，我们将使用平方误差将 `Y` 与卷积层的输出进行比较。然后我们可以计算渐变来更新内核。为了简单起见，在下面我们使用二维卷积层的内置类，并忽略偏差。
+我们先构造一个卷积层，并将其卷积核初始化为随机张量。接下来，在每次迭代中，我们比较  `Y` 与卷积层的输出的平方误差，然后计算梯度来更新内核。为了简单起见，我们在此使用内置的二维卷积层，并忽略偏差。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:186]: 第一句建议翻译一下

Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:186]: 卷积层的输出-> 去掉“的”。否则不太好理解是 “`Y` 与卷积层”的输出 还是 `Y` 与”卷积层“的输出

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 12]
```diff
@@ -168,19 +181,19 @@ corr2d(d2l.transpose(X), K)
...
-# Construct a two-dimensional convolutional layer with 1 output channel and a
-# kernel of shape (1, 2). For the sake of simplicity, we ignore the bias here
+# 构造一个二维卷积层，它具有1个输出通道和形状为（1，2）的卷积核
 conv2d = nn.Conv2D(1, kernel_size=(1, 2), use_bias=False)
 conv2d.initialize()
 
-# The two-dimensional convolutional layer uses four-dimensional input and
-# output in the format of (example, channel, height, width), where the batch
-# size (number of examples in the batch) and the number of channels are both 1
+# 这个二维卷积层使用四维输入和输出格式（例如，批量大小、通道、高度、宽度），
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:193]: 去掉例如。

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 13]
```diff
@@ -236,16 +245,16 @@ for i in range(10):
...
         weights = conv2d.get_weights()
         weights[0] = conv2d.weights[0] - update
         conv2d.set_weights(weights)
         if (i + 1) % 2 == 0:
-            print(f'batch {i + 1}, loss {tf.reduce_sum(l):.3f}')
+            print(f'batch {i+1}, loss {tf.reduce_sum(l):.3f}')
 ```
 
-请注意，在 10 次迭代之后，错误已降至一个小值。现在我们来看看我们学到的内核张量。
+在 $10$ 次迭代之后，误差已经骤降。现在我们来看看我们所学的卷积核的权重张量。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:257]: 降到足够低

File: chapter_convolutional-neural-networks/conv-layer.md [hunk 14]
```diff
@@ -261,44 +270,58 @@ d2l.reshape(conv2d.weight.data, (1, 2))
...
+
 
-值得注意的是，由于内核是从深度学习中的数据中学习的，因此无论这些层执行严格的卷积操作还是交叉相关运算，卷积层的输出都不受影响。
 
-为了说明这一点，假设一个卷积层执行 * 交叉关联 * 并在 :numref:`fig_correlation` 中学习内核，此处表示为矩阵 $\mathbf{K}$。假设其他条件保持不变，当此层执行严格的 * 卷积 * 时，在 $\mathbf{K}'$ 水平和垂直翻转后，学习的内核 $\mathbf{K}'$ 将与 $\mathbf{K}$ 相同。也就是说，当卷积层对 :numref:`fig_correlation` 和 $\mathbf{K}'$ 中的输入执行严格的 * 卷积 * 时，将获得 :numref:`fig_correlation` 中的相同输出（输入和 $\mathbf{K}$ 的互相关）。
+## 特征映射和感受野
 
-为了与深度学习文献中的标准术语保持一致，我们将继续将互相关运算称为卷积，尽管严格来说，它略有不同。此外，我们使用术语 *element* 来指代表层表示或卷积内核的任何张量的条目（或组件）。
+如在 :numref:`subsec_why-conv-channels` 中所述， :numref:`fig_correlation` 中输出的卷积层有时被称为 *特征映射* （Feature Map），因为它可以被视为一个输入映射到下一层的空间维度的转换器。
+在CNN中，对于某一层的任何元素 $x$ ，其 *感受野* （Receptive Field）是指在前向传播期间可能影响 $x$ 计算的所有元素（来自所有先前层）。
```
Comment (xiaotinghe) [inline chapter_convolutional-neural-networks/conv-layer.md:295]: 任何->任意

**Comment by mli** [comment]
Job d2l-zh/PR-589/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/

File: chapter_convolutional-neural-networks/conv-layer.md (hunk 6 above)
Comment (goldmermaid) [reply chapter_convolutional-neural-networks/conv-layer.md:158]: fixed all

**Comment by mli** [comment]
Job d2l-zh/PR-589/11 is complete. 
Check the results at http://preview.d2l.ai/d2l-zh/PR-589/
